ios/Classes/Embeds/AppcuesFrameViewFactory.swift

```diff
@@ -0,0 +1,195 @@
+        frame: CGRect,
+        viewIdentifier viewId: Int64,
+        arguments args: Any?,
+        binaryMessenger messenger: FlutterBinaryMessenger?
+    ) {
+        let args = args as? [String: Any]
+
+        _view = WrapperView(
+            plugin: plugin,
+            binaryMessenger: messenger,
+            frameID: args?["frameId"] as? String,
+            viewIdentifier: viewId
+        )
+        super.init()
+    }
+
+    func view() -> UIView {
+        return _view
+    }
+}
+
+
+class WrapperView: UIView, FlutterStreamHandler {
+
+
+    private let plugin: SwiftAppcuesFlutterPlugin
+    private var eventChannel: FlutterEventChannel?
+    private var eventSink: FlutterEventSink?
+
+    var frameID: String? = nil
+
+    private weak var frameViewController: AppcuesFrameVC?
+
+    init(
+        plugin: SwiftAppcuesFlutterPlugin,
+        binaryMessenger messenger: FlutterBinaryMessenger?,
+        frameID: String?,
+        viewIdentifier viewId: Int64
+    ) {
+        self.plugin = plugin
+        self.frameID = frameID
+
+        if let messenger = messenger {
+            eventChannel = FlutterEventChannel(name: "com.appcues.flutter/frame/\(viewId)", binaryMessenger: messenger)
+        }
+
+        super.init(frame: .zero)
+
+        eventChannel?.setStreamHandler(self)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if let frameViewController = frameViewController {
+            frameViewController.view.frame = bounds
+        } else {
+            setupFrame()
+        }
+    }
+
+    private func setupFrame() {
+        guard let parentVC = parentViewController,
+              let frameID = frameID else {
+            return
+        }
+
+        let frameVC = AppcuesFrameVC(parentView: self)
+        parentVC.addChild(frameVC)
+        addSubview(frameVC.view)
+        frameVC.view.frame = bounds
+        frameVC.didMove(toParent: parentVC)
+        self.frameViewController = frameVC
+
+        plugin.register(frameID: frameID, for: frameVC.frameView, on: frameVC)
+    }
+
+    func setIntrinsicSize(preferredContentSize: CGSize, isHidden: Bool) {
+        let size: CGSize
+
+        if isHidden {
+            // When the frame is hidden, size this WrapperView to 0.
+            size = .zero
+        } else {
+            if preferredContentSize == .zero {
+                // When the frame is NOT hidden and the current size is 0, set a non-zero height,
+                // which allows the content to start its layout algorithm and determine the required size.
+                size = CGSize(width: 0, height: 1)
+            } else {
+                // When the frame is NOT hidden use the size calculated by the content.
+                size = preferredContentSize
+            }
+        }
+
+        eventSink?(size.height)
```

Comment:
currently just communicating the `height` and assuming the `width` is controlled by the host application - either full width or some fixed constrained value on the embed frame on the Flutter side. Need to validate this.